src/cloudflare/internal/ai-api.ts

```diff
@@ -52,37 +59,36 @@ export class Ai {
     const body = JSON.stringify({
       inputs,
       options: {
-        debug: this.options?.debug,
+        debug: this.options.debug,
       },
     });
 
     const fetchOptions = {
       method: "POST",
       body: body,
       headers: {
-        ...(this.options?.sessionOptions?.extraHeaders || {}),
-        ...(this.options?.extraHeaders || {}),
+        ...(this.options.sessionOptions?.extraHeaders || {}),
+        ...(this.options.extraHeaders || {}),
         "content-type": "application/json",
-        // 'content-encoding': 'gzip',
         "cf-consn-sdk-version": "2.0.0",
         "cf-consn-model-id": `${this.options.prefix ? `${this.options.prefix}:` : ""}${model}`,
       },
     };
 
-    const res = await this.fetcher.fetch("http://workers-binding.ai/run?version=2", fetchOptions);
+    const res = await this.fetcher.fetch("http://workers-binding.ai/run?version=3", fetchOptions);
 
     this.lastRequestId = res.headers.get("cf-ai-req-id");
 
     if (inputs['stream']) {
       if (!res.ok) {
-        throw new InferenceUpstreamError(await res.text());
+        throw await this._parseError(res)
```

Comment:
Has backwards compat been considered here? Are there any possible cases where people would previously be expecting a specific error string (to retry, log, etc.) and this might now change?